File: leetcode/206_reverse_linked_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *prev = nullptr;
        ListNode *front = head;
        ListNode *tmp = head;
        while (front != nullptr) {
            tmp = tmp->next;
            front->next = prev;
            prev = front;
            front = tmp;
        }
        return prev;
    }
};

File: leetcode/206_reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        front = head
        temp = head
        while front:
            temp = front.next
            front.next = prev
            prev = front
            front = temp
        
        return prev